primitives/types/dispatch_errors.go

```diff
@@ -140,14 +176,37 @@ func DecodeDispatchError(buffer *bytes.Buffer) (DispatchError, error) {
 // CustomModuleError A custom error in a module.
 type CustomModuleError struct {
 	Index   sc.U8             // Module index matching the metadata module index.
-	Error   sc.U32            // Module specific error value.
+	Err     sc.U32            // Module specific error value.
 	Message sc.Option[sc.Str] // Varying data type Option (Definition 190). The optional value is a SCALE encoded byte array containing a valid UTF-8 sequence.
 }
 
-func (e CustomModuleError) Encode(buffer *bytes.Buffer) error {
+func (err CustomModuleError) Error() string {
+	switch sc.U8(err.Err) {
+	case 0: // ErrorVestingBalance
+		return "Vesting balance too high to send value"
+	case 1: // ErrorLiquidityRestrictions
+		return "Account liquidity restrictions prevent withdrawal"
+	case 2: // ErrorInsufficientBalance
+		return "Balance too low to send value"
+	case 3: // ErrorExistentialDeposit
+		return "Value too low to create account due to existential deposit"
+	case 4: // ErrorKeepAlive
+		return "Transfer/payment would kill account"
+	case 5: // ErrorExistingVestingSchedule
+		return "A vesting schedule already exists for this account"
+	case 6: // ErrorDeadAccount
+		return "Beneficiary account must pre-exist"
+	case 7: // ErrorTooManyReserves
+		return "Number of named reserves exceed MaxReserves"
```

Comment:
Let's log the values of the struct, as these values (0-7) are tightly coupled to the balances module.